plugins/source/postgresql/client/cdc.go

```diff
@@ -0,0 +1,289 @@
+package client
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/jackc/pglogrepl"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgconn"
+	"github.com/jackc/pgx/v5/pgproto3"
+	"github.com/jackc/pgx/v5/pgtype"
+)
+
+func (c *Client) tablesWithPks() []string {
+	var tables []string
+	for _, table := range c.Tables {
+		if len(table.PrimaryKeys()) > 0 {
+			tables = append(tables, table.Name)
+		}
+	}
+	return tables
+}
+
+func (c *Client) createPublicationForTables(ctx context.Context, conn *pgconn.PgConn) error {
+	tables := c.tablesWithPks()
+	sql := fmt.Sprintf("CREATE PUBLICATION %s FOR TABLE %s", pgx.Identifier{c.spec.Name}.Sanitize(), strings.Join(tables, ","))
+	reader := conn.Exec(ctx, sql)
+	if _, err := reader.ReadAll(); err != nil {
+		var pgErr *pgconn.PgError
+		if !errors.As(err, &pgErr) {
+			// not recoverable error
+			return fmt.Errorf("failed to create publication: %w", err)
+		}
+		if pgErr.Code != "42710" {
```

Comment:
```suggestion
                // 42710 means publication already exists
		if pgErr.Code != "42710" {
```

Maybe add a comment or make a function `isPublicationExistsError` for readability